components/api/src/main/java/com/hotels/styx/api/messages/StreamingHttpRequest.java

```diff
@@ -0,0 +1,693 @@
+     */
+    public HttpMethod method() {
+        return method;
+    }
+
+    /**
+     * Returns the requested URI (or alternatively, path).
+     *
+     * @return The URI being requested
+     */
+    public Url url() {
+        return url;
+    }
+
+    /**
+     * Returns the requested path.
+     *
+     * @return the path being requested
+     */
+    public String path() {
+        return url.path();
+    }
+
+    /**
+     * Returns {@code true} if and only if the connection can remain open and thus 'kept alive'.
+     * This methods respects the value of the {@code "Connection"} header first and if this has no such header
+     * then the keep-alive status is determined by the HTTP version, that is, HTTP/1.1 is keep-alive by default,
+     * HTTP/1.0 is not keep-alive by default.
+     *
+     * @return true if the connection is keep-alive
+     */
+    public boolean keepAlive() {
+        return HttpMessageSupport.keepAlive(headers, version);
+    }
+
+    /**
+     * Checks if the request has been transferred over a secure connection. If the protocol is HTTPS and the
+     * content is delivered over SSL then the request is considered to be secure.
+     *
+     * @return true if the request is transferred securely
+     */
+    public boolean isSecure() {
+        return secure;
+    }
+
+    /**
+     * Returns the remote client address that initiated the current request.
+     *
+     * @return the client address for this request
+     */
+    public InetSocketAddress clientAddress() {
+        return this.clientAddress;
+    }
+
+    /**
+     * Get a query parameter by name if present.
+     *
+     * @param name parameter name
+     * @return query parameter if present
+     */
+    public Optional<String> queryParam(String name) {
+        return url.queryParam(name);
+    }
+
+    /**
+     * Gets query parameters by name.
+     *
+     * @param name parameter name
+     * @return query parameters
+     */
+    public Iterable<String> queryParams(String name) {
+        return url.queryParams(name);
+    }
+
+    /**
+     * Get all query parameters.
+     *
+     * @return all query parameters
+     */
+    public Map<String, List<String>> queryParams() {
+        return url.queryParams();
+    }
+
+    /**
+     * Get the names of all query parameters.
+     *
+     * @return the names of all query parameters.
+     */
+    public Iterable<String> queryParamNames() {
+        return url.queryParamNames();
+    }
+
+    /**
+     * Return a new {@link Builder} that will inherit properties from this request.
+     * This allows a new request to be made that will be identical to this one except for the properties
+     * overridden by the builder methods.
+     *
+     * @return new builder based on this request
+     */
+    public Builder newBuilder() {
+        return new Builder(this);
+    }
+
+    public <T> Observable<FullHttpRequest<T>> toFullHttpRequest(Function<ByteBuf, T> decoder, int maxContentBytes) {
```

Comment:
See the current implementation of `FullHttpRequest`. For the latest developments consult the master branch. It has been merged as a part of PR #50.

In effect the `decoder` argument can now be removed.